core/src/main/java/org/mindtrails/controller/AccountController.java

```diff
@@ -87,7 +89,19 @@ private  void addAttributesForCreateParticipantForm(ModelMap model) {
     }
 
     @RequestMapping(value="create", method = RequestMethod.GET)
-    public String createForm (ModelMap model, HttpSession session) {
+    public String createForm (ModelMap model, 
+                              HttpServletRequest request,
+                              HttpSession session) {
+        
+        String requestURI = request.getRequestURI();
+
+        // Prevent joining the Kaiser study without a randomization condition
+        // Kaiser study has a separate RequestMapping setup that handles query params (e.g. when condition is specified)
+        // This shouldn't be needed, but is an extra layer of protection in case someone finds the account/create url
+        if ((requestURI.contains("kaiser")) && ((String)session.getAttribute("condition") == null)) {
```

Comment:
Rather than modify this code to raise an error out of core, I would instead call use the KaiserParticipantService (which is loaded below via participant Service, and override it's "isEligible" method to instead raise an error if not condition comes in.  and handle it all that way, with no custome code in this account controller at all.